packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -75,7 +100,7 @@ export function useComboBoxState<T extends object>(props: ComboBoxStateProps<T>)
   };
 
   let isInitialRender = useRef(true);
-  let lastSelectedKey = useRef(null);
+  let lastSelectedKey = useRef(props.selectedKey || props.defaultSelectedKey || null);
```

Comment:
Regarding the question concerning this line and 
```
    if (isInitialRender.current && (props.inputValue === undefined && props.defaultInputValue === undefined)) {
      resetInputValue();
    }
```
This change fixes two bugs. 
1. The first bug is that ComboBox with `isOpen: true` and `selectedKey/defaultSelectedKey: some key` calls `isOpenChange(false)` on initial render. If `lastSelectedKey = useRef(null)`, the useEffect calls `triggerState.close()` due to this block of code:
```
    // Close when an item is selected, if open state or selectedKey is uncontrolled.
    if (
      selectedKey != null &&
      selectedKey !== lastSelectedKey.current &&
      (props.isOpen === undefined || props.selectedKey === undefined)
    ) {
      triggerState.close();
    }
```
hence the change to `let lastSelectedKey = useRef(props.selectedKey || props.defaultSelectedKey || null);`

2. The second bug occurs due to the above change where now ComboBoxes that have `selectedKey/defaultSelectedKey` render with empty input fields on initial render. This used to work when `lastSelectedKey` was initialized to `null` because it would hit this block of code in the useEffect, calling `resetInputValue`:
```
    if (
      selectedKey !== lastSelectedKey.current &&
      (props.inputValue === undefined || props.selectedKey === undefined) &&
      !(isInitialRender.current && props.defaultInputValue !== undefined)
    ) {
      resetInputValue();
    } else {
      lastValue.current = inputValue;
    }
```
To compensate for the fact that `lastSelectedKey` now initializes as `defaultSelectedKey/selectedKey` if provided, I added
```
    // If it is the intial render and inputValue isn't controlled nor has an intial value, set input to match current selected key if any
    if (isInitialRender.current && (props.inputValue === undefined && props.defaultInputValue === undefined)) {
      resetInputValue();
    }
```

Reply:
ok makes sense. should we do this instead to account for falsey keys?

```suggestion
  let lastSelectedKey = useRef(props.selectedKey ?? props.defaultSelectedKey ?? null);
```